examples/encoder-decoder-coupling.lua

```diff
@@ -31,12 +31,13 @@ function main()
   opt.inputSeqLen = 3 -- length of the encoded sequence
 
   -- Some example data
-  local encInSeq, decInSeq, decOutSeq = torch.Tensor({1,2,3}), torch.Tensor({1,2,3,4}), torch.Tensor({2,3,4,1})
-
+  local encInSeq, decInSeq, decOutSeq = torch.Tensor({{1,2,3},{3,2,1}}), torch.Tensor({{1,2,3,4},{4,3,2,1}}), torch.Tensor({{2,3,4,1},{1,2,4,3}})
+  decOutSeq = nn.SplitTable(2):forward(decOutSeq)
+  
   -- Encoder
   local enc = nn.Sequential()
   enc:add(nn.LookupTable(opt.vocabSz, opt.hiddenSz))
-  enc:add(nn.SplitTable(1))
+  enc:add(nn.SplitTable(2)) --Split along the second axis, where the first axis is the mini-batch
```

Comment:
@cheng6076 Can you change this from `nn.SplitTable(2)` to `nn.SplitTable(1,1)`. That way, the module works for both online and mini-batch mode. https://github.com/torch/nn/blob/master/doc/table.md#nn.SplitTable . Same for line 48.